### PR TITLE
KTOR-3968 update Compression API docs

### PR DESCRIPTION
**Pull request by andreyaksenov**

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt [hunk 1]
```diff
@@ -43,7 +43,22 @@ private object ContentEncoding : Hook<suspend ContentEncoding.Context.(Applicati
 }
 
 /**
- * A plugin for compressing a response based on specific conditions and the ability of the client to decompress it
+ * A plugin that provides the capability to compress outgoing content.
```
Comment (Stexxe) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt:46]: It's to replace `outgoing content` with `response body`

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt [hunk 2]
```diff
@@ -43,7 +43,22 @@ private object ContentEncoding : Hook<suspend ContentEncoding.Context.(Applicati
...
+ * You can use different compression algorithms, including `gzip` and `deflate`,
+ * specify the required conditions for compressing data (such as a content type or response size),
+ * or even compress data based on specific request parameters.
+ *
+ * The example below shows how to compress JavaScript using `gzip` with the specified priority:
```
Comment (Stexxe) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt:51]: Javascript what?

Comment (andreyaksenov) [reply ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Compression.kt:51]: > Javascript is a programming language.

thanks

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt [hunk 3]
```diff
@@ -24,41 +24,41 @@ public data class CompressionOptions(
 )
 
 /**
- * Configuration for an encoder
+ * An encoder configuration
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:27]: For [Compression] plugin

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt [hunk 4]
```diff
@@ -170,36 +170,36 @@ public fun CompressionConfig.deflate(block: CompressionEncoderBuilder.() -> Unit
 }
 
 /**
- * Appends `identity` encoder
+ * Appends the `identity` encoder with the [block] configuration
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:173]: .

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt [hunk 5]
```diff
@@ -153,14 +153,14 @@ public class CompressionEncoderBuilder internal constructor(
 }
 
 /**
- * Appends `gzip` encoder
+ * Appends the `gzip` encoder with the [block] configuration
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:156]: .

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt [hunk 6]
```diff
@@ -170,36 +170,36 @@ public fun CompressionConfig.deflate(block: CompressionEncoderBuilder.() -> Unit
...
 public fun ConditionsHolderBuilder.minimumSize(minSize: Long) {
     condition { content -> content.contentLength?.let { it >= minSize } ?: true }
 }
 
 /**
- * Appends a content type condition to the encoder or Compression configuration.
+ * Appends a content type condition to the encoder or the [Compression] configuration.
  *
- * Please note that adding a single match condition removes the default configuration.
+ * Note that adding a single condition removes the default configuration.
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:202]: [default]

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Encoders.kt [hunk 7]
```diff
@@ -10,11 +10,11 @@ import kotlinx.coroutines.*
...
 /**
- * Represents a Compression encoder
+ * A [Compression] encoder
  */
 public interface CompressionEncoder {
     /**
-     * May predict compressed length based on the [originalLength] or return `null` if it is impossible.
+     * Tries to estimate a compressed length based on the [originalLength] or return `null` if it's impossible
      */
     public fun predictCompressedLength(originalLength: Long): Long? = null
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Encoders.kt:19]: Provide estimation for compressed length or null ...

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt [hunk 8]
```diff
@@ -170,36 +170,36 @@ public fun CompressionConfig.deflate(block: CompressionEncoderBuilder.() -> Unit
...
- * Appends `identity` encoder
+ * Appends the `identity` encoder with the [block] configuration
  */
 public fun CompressionConfig.identity(block: CompressionEncoderBuilder.() -> Unit = {}) {
     encoder("identity", IdentityEncoder, block)
 }
 
 /**
- * Appends a custom condition to the encoder or Compression configuration.
+ * Appends a custom condition to the encoder or the [Compression] configuration.
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:180]: Add a [predicate] to check if the content should use compression

**Review by e5l — APPROVED** [review]
lgtm, please check some small stuff

File: ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt (hunk 6 above)
Comment (andreyaksenov) [reply ktor-server/ktor-server-plugins/ktor-server-compression/jvm/src/io/ktor/server/plugins/compression/Config.kt:202]: > [default]

IDEA shows "Cannot resolve symbol 'default'".

**Pull request merged**